### PR TITLE
manage friends from brand portal

### PR DESCRIPTION
**Pull request by danigargar**
#### Type Of Change 
- [ ] Small bug fix
- [X] New feature or enhancement
- [ ] Breaking change

#### Checklist:
- [X] Commits are named and have tag following [commit rules](https://github.com/irontec/ivozprovider/blob/bleeding/doc/dev/en/commits.md)
- [X] Commits are split per component (schema, web/admin, kamusers, agis, ..)
- [X] Changes have been tested locally
- [ ] Fixes an existing issue (Fixes #XXXX) 
- [ ] Upport from existing Pull request #XXXX

### CODE REVIEW

**Review by ironArt3mis — CHANGES_REQUESTED** [review]
Functional review required

File: web/portal/brand/src/entities/Friend/hooks/useIntercompanyId.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    string
+  > | null>(null);
+  const [, cancelToken] = useCancelToken();
+
+  const isInterVpbx = formik.values.directConnectivity === 'intervpbx';
+  const companyId = formik.values.company;
+  const interCompanyId = formik.values.interCompany;
+
+  useEffect(() => {
+    setInterCompany([]);
```
Comment (mmadariaga) [inline web/portal/brand/src/entities/Friend/hooks/useIntercompanyId.tsx:25]: An array is not a `Record<number, string>`

File: web/portal/brand/src/entities/Friend/hooks/useIntercompanyId.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    if (!companyId) {
+      return;
+    }
+
+    const includeId = edit ? interCompanyId : null;
+
+    UnassignedCompanySelectOptions(
+      {
+        callback: (options) => {
+          setInterCompany(options || []);
```
Comment (mmadariaga) [inline web/portal/brand/src/entities/Friend/hooks/useIntercompanyId.tsx:40]: Beware the types

File: web/rest/brand/features/provider/friend/postFriend.feature [hunk 3]
```diff
@@ -4,8 +4,8 @@ Feature: Create friends
   I need to be able to create them through the API.
 
   @createSchema
-  Scenario: Create a friend
-    Given I add Brand Authorization header
+  Scenario: Create a friend without authorization
+    Given I add Authorization header
```
Comment (mmadariaga) [inline web/rest/brand/features/provider/friend/postFriend.feature:8]: Unnecessary I guess

File: web/rest/brand/src/Controller/Provider/CompaniesByCorporateUnassignedAction.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            );
+
+            if (count($friends) !== 0) {
+                continue;
+            }
+
+            $response[] = $interCompany;
+        }
+
+        return $response;
```
Comment (mmadariaga) [inline web/rest/brand/src/Controller/Provider/CompaniesByCorporateUnassignedAction.php:76]: It would be nice to extract this into an application service

**Review by mmadariaga — CHANGES_REQUESTED** [review]

File: web/portal/brand/src/entities/Friend/Field/StatusIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    return (
+      <Tooltip title={_('Inter client connectivity')}>
+        <StyledStatusIconRotateLeftIcon />
+      </Tooltip>
+    );
+  }
+
+  //@TODO POSPONED else RegisterStatus::getLocationStatus
```
[the remark this answers is not in the document]
Comment (danigargar) [reply web/portal/brand/src/entities/Friend/Field/StatusIcon.tsx:58]: Copy/paste from web/portal/client/src/entities/Friend/Field/StatusIcon.tsx

I remove it here.

**Review by mmadariaga — APPROVED** [review]

**Review by cruzccl — APPROVED** [review]

**Review by ironArt3mis — APPROVED** [review]

**Pull request merged**